tests/conftest.py

```diff
@@ -7310,28 +7310,39 @@ def factory(namespace=None):
             pvk_obj = PVKeyrotation(sc_obj)
             pvk_obj.annotate_storageclass_key_rotation(schedule="*/3 * * * *")
 
-        # Load VMconfigs from cnv_vm_workload yaml
+        # Load VM configs from cnv_vm_workload yaml
         vm_configs = templating.load_yaml(constants.CNV_VM_WORKLOADS)
 
-        # Loop through vm_configs and create the VMs using the cnv_workload fixture
-        for index, vm_config in enumerate(vm_configs["cnv_vm_configs"]):
-            # Determine the storage class based on the compression type
-            if vm_config["sc_compression"] == "default":
-                storageclass = sc_obj_def_compr.name
-            elif vm_config["sc_compression"] == "aggressive":
-                storageclass = sc_obj_aggressive.name
-            vm_obj = cnv_workload(
-                volume_interface=vm_config["volume_interface"],
-                access_mode=vm_config["access_mode"],
-                storageclass=storageclass,
-                pvc_size="30Gi",  # Assuming pvc_size is fixed for all
-                source_url=constants.CNV_FEDORA_SOURCE,  # Assuming source_url is the same for all VMs
-                namespace=namespace,
-            )[index]
-            if vm_config["sc_compression"] == "aggressive":
-                vm_list_agg_compr.append(vm_obj)
-            else:
-                vm_list_default_compr.append(vm_obj)
+        # Use ThreadPoolExecutor to create VMs parallel
+        with ThreadPoolExecutor() as executor:
+            futures = []
```

Comment:
Looks like you are storing some metadata, should dict be used here?

Reply:
Done